hedera-node/src/main/java/com/hedera/services/state/virtual/UniqueTokenKey.java

```diff
@@ -0,0 +1,152 @@
+package com.hedera.services.state.virtual;
+
+import com.swirlds.common.io.SerializableDataInputStream;
+import com.swirlds.common.io.SerializableDataOutputStream;
+import com.swirlds.virtualmap.VirtualKey;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * Represents a key for a unique token (NFT).
+ */
+public class UniqueTokenKey implements VirtualKey<UniqueTokenKey> {
+	private static final long CLASS_ID = 0x17f77b311f6L;
+
+	/** Current version of the encoding scheme. */
+	/* package */ static final int CURRENT_VERSION = 1;
+
+	/**
+	 * Expected maximum number of bytes this class will serialize to. Serialization will format will be:
+	 *  - (1 byte) number of bytes, 1-255, following this byte that will contain the token number.
```

Comment:
In HTS parlance, a "token number" is the entity number for a token type. For example, the USDC token on mainnet is entity `0.0.456858`---a fungible token type with token number 456858 .

Then an NFT is a particular **serial number** that has been minted for a non-fungible token type. For example, I am the proud owner of HBot serial number 803 (NFI 😉); where the HBot token type is `0.0.697893` and my NFT would commonly be written as `0.0.697893.803`.

Now, the [current `MerkleMap` key](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/ServicesState.java#L358) captures both the token number _and_ the serial number by packing them into a single `long` in an `EntityNumPair`; where the high-order 32-bits are interpreted as an unsigned `long` token number, and the low-order 32-bits are an unsigned `long` serial number. (So right now we are limited to ~4B token types, and ~4B serial numbers per token type.)

Anyways, all that to say, @jasperpotts [has advised](https://github.com/hashgraph/hedera-services/issues/2960) we move to using the full `long` for entity numbers. So imo we want `UniqueTokenKey` to be able to represent at least `Long.BYTES + Integer.BYTES + 1` of information (a `long` token number + an `int` serial number).

Reply:
I don't know of a use case that requires minting 4B NFTs, but hopefully we would have time to switch to our representation well before the limit was hit 😁.

Reply:
Spoke to Jasper and he confirmed that the token serial should be a long.